akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -847,11 +847,10 @@ private[akka] class ActorSystemImpl(
   val internalCallingThreadExecutionContext: ExecutionContext =
     dynamicAccess
       .getObjectFor[ExecutionContext]("scala.concurrent.Future$InternalCallbackExecutor$")
-      .getOrElse(new ExecutionContext with BatchingExecutor {
-        override protected def unbatchedExecute(r: Runnable): Unit = r.run()
-        override protected def resubmitOnBlock: Boolean = false // Since we execute inline, no gain in resubmitting
-        override def reportFailure(t: Throwable): Unit = dispatcher.reportFailure(t)
-      })
+      .getOrElse(
+        dynamicAccess
+          .getObjectFor[ExecutionContext]("scala.concurrent.ExecutionContext$parasitic$")
+          .getOrElse(ExecutionContexts.sameThreadExecutionContext))
```

Comment:
Given the test, this is a fallback that should be rarely hit, right? Makes more sense than the custom anonymous `BatchingExecutor` subclass in any case :+1: